src/main/clm_driver.F90

```diff
@@ -1137,7 +1141,7 @@ subroutine clm_drv(doalb, nextsw_cday, declinp1, declin, rstwr, nlend, rdate, ro
                temperature_inst%t_ref2m_patch, temperature_inst%t_soisno_col)
        end if
 
-       call cnfire_method_inst%UpdateAccVars(bounds_proc)
+       call fates_fire_data_inst%UpdateAccVars(bounds_proc)
```

Comment:
@ekluzek 
I'm afraid I don't see how this line will work without an `if` statement like this:

`if (use_fates_spitfire > 1) then`
`call fates_fire_data_inst%UpdateAccVars(bounds_proc)`
`else`
`call fates_fire_no_data_inst%UpdateAccVars(bounds_proc)`

The same comment goes for the corresponding lines in `clm_initializeMod` and `clm_instMod`.

Reply:
@ekluzek 
Am I still missing new code that links these two options to the `fates_fire_data_method`? According to the template that I see for routines like CNFireArea, for example? Does all this infrastructure serve to avoid `if` statements and improve performance?

Reply:
I'm not seeing what you are missing offhand. And we should be able to get this to work how we want, because we use it elsewhere. Let me checkout your code and see if I can see what's missing.

What test case are you trying? And do you have everything needed checked in at this point?

Reply:
The issue here is that there needs to be a subroutine in clmfates_interface that does the UpdateAccVars for fire (since the fire data instance is inside of it). This is something I neglected to notice the other day, sorry about that.